starlark/int.go

```diff
@@ -8,74 +8,77 @@ import (
 	"fmt"
 	"math"
 	"math/big"
+	"strconv"
 
 	"go.starlark.net/syntax"
 )
 
 // Int is the type of a Starlark int.
-type Int struct{ bigint *big.Int }
+type Int struct {
+	// big != nil <=> value is not representable as int32.
+	// We use only the signed 32 bit range of small to ensure
+	// that small+small and small*small do not overflow.
+
+	small int64    // minint32 <= small <= maxint32
+	big   *big.Int // big.BitLen() > 32
```

Comment:
You're right that > 32 isn't correct, and the BitLen alone is not the right predicate to express the boundary between big and small.  Let's update it to:

// big.BitLen() >= 32 (note: MinInt32 is sole small value of BitLen 32)

Sorry for the bad directions.

Reply:
No problem, fixed the minInt32 edge case.